dask/dataframe/io/tests/test_parquet.py

```diff
@@ -152,3 +152,42 @@ def test_ordering():
 
         ddf2 = read_parquet(tmp, index='myindex')
         assert_eq(ddf, ddf2)
+
+
+@pytest.mark.parametrize('df,kwargs', [
+    (pd.DataFrame({'x': [3, 2, 1]}), {}),
+    (pd.DataFrame({'x': ['c', 'a', 'b']}), {'object_encoding': 'utf8'}),
+    (pd.DataFrame({'x': ['cc', 'a', 'bbb']}), {'object_encoding': 'utf8'}),
+    (pd.DataFrame({'x': [b'a', b'b', b'c']}), {'object_encoding': 'bytes'}),
+    (pd.DataFrame({'x': pd.Categorical(['a', 'b', 'a'])}), {'object_encoding': 'utf8'}),
+    (pd.DataFrame({'x': pd.Categorical([1, 2, 1])}), {}),
+    (pd.DataFrame({'x': list(map(pd.Timestamp, [3000, 2000, 1000]))}), {}),
+    (pd.DataFrame({'x': [3000, 2000, 1000]}).astype('M8[ns]'), {}),
+    pytest.mark.xfail((pd.DataFrame({'x': [3, 2, 1]}).astype('M8[ns]'), {}),
+                      reason="Parquet doesn't support nanosecond precision"),
+    (pd.DataFrame({'x': [3, 2, 1]}).astype('M8[us]'), {}),
+    (pd.DataFrame({'x': [3, 2, 1]}).astype('M8[ms]'), {}),
+    (pd.DataFrame({'x': [3, 2, 1]}).astype('uint16'), {}),
+    (pd.DataFrame({'x': [3, 2, 1]}).astype('float32'), {}),
+    (pd.DataFrame({'x': [3, 1, 2]}, index=[3, 2, 1]), {}),
+    (pd.DataFrame({'x': [3, 1, 5]}, index=pd.Index([1, 2, 3], name='foo')), {}),
+    (pd.DataFrame({'x': [1, 2, 3],
+                  'y': [3, 2, 1]}), {}),
+    (pd.DataFrame({'x': [1, 2, 3],
+                  'y': [3, 2, 1]}, columns=['y', 'x']), {}),
+    (pd.DataFrame({'0': [3, 2, 1]}), {}),
+    (pd.DataFrame({'x': [3, 2, None]}), {}),
+    (pd.DataFrame({'-': [3., 2., None]}), {}),
+    (pd.DataFrame({'.': [3., 2., None]}), {}),
+    (pd.DataFrame({' ': [3., 2., None]}), {}),
+])
+def test_roundtrip(df, kwargs):
+    with tmpdir() as tmp:
+        tmp = str(tmp)
+        if df.index.name is None:
+            df.index.name = 'index'
+        ddf = dd.from_pandas(df, npartitions=2)
+
+        to_parquet(tmp, ddf, **kwargs)
+        ddf2 = read_parquet(tmp, index=df.index.name)
```

Comment:
You would need `categories=` to get the dtype right, but you can add `check_categorical=False` to the assert_equal, that I think should check the actual values.